pr-db-mongo/pr_db_mongo.py

```diff
@@ -0,0 +1,106 @@
+import subprocess
+import sys
+from urllib.parse import quote_plus, urlsplit
+
+PLUGIN_NAME = "pr-db-mongo"
+
+
+def execute_bash(command):
+    return subprocess.run(command, check=True, capture_output=True, text=True)
+
+
+def get_uri(db_url, db_name=""):
+    split_url = urlsplit(db_url)
+    username = split_url.username
+    password = split_url.password
+    scheme = split_url.scheme
+    hostname = split_url.hostname
+    port = split_url.port
+    query = split_url.query
+    credentials = ":".join([quote_plus(c) for c in (username, password)])
+    host = f"{hostname}:{port}" if port else hostname
+    options = f"/{db_name}?{query}" if query else ""
+
+    return f"{scheme}://{credentials}@{host}{options}"
+
+
+def configure_pr_app(original_mongodb_url, app_name):
+    mongodb_url = get_uri(original_mongodb_url, app_name)
+    commands = [
+        ["dokku", "config:set", "--no-restart", app_name, f"MONGODB_URL={mongodb_url}"],
+        # TODO: get main domain from original app and modify it
+        ["dokku", "domains:add", app_name, f"{app_name}.dev.codeforafrica.org"],
+        ["dokku", "letsencrypt:enable", app_name],
+    ]
+    for command in commands:
+        execute_bash(command)
+
+
+def clone_pr_database(original_mongodb_url, app_name):
+    split_url = urlsplit(original_mongodb_url)
+    source = split_url.path.lstrip("/")
+    uri = get_uri(original_mongodb_url)
+    command = [
+        "mongodump",
+        "-vvvvv",
+        "--archive",
+        f"--uri={uri}",
+        f"--db={source}",
+        "|",
+        "mongorestore",
+        "-vvvvv",
+        f"--uri={uri}",
```

Comment:
Does the same uri work for `mongodump` and `mongorestore` @koechkevin ?

Reply:
Yes,

in this case it is the URI without database.